### ISSUE TITLE
Fix parsing of compatible release clause

### ISSUE THREAD

**Issue by ddelange** [PyO3/maturin-action#286]
Setting `maturin~=1.0` ([compatible release clause](https://peps.python.org/pep-0440/#compatible-release) equivalent to `maturin>=1.0,<2`) in pyproject.toml will result in maturin-action [picking](https://github.com/ddelange/blake3-py/actions/runs/11001099987/job/30545001107#step:4:19) v1.0.1 instead of v1.7.1.

Quick fix for users: https://github.com/ddelange/blake3-py/pull/1/commits/2d56b535f32ae61bffb80d974db8bb039bebf56e 

Found during development: https://github.com/oconnor663/blake3-py/pull/45

**Comment by messense** [comment]
This is actually a known issue because the version spec in Python isn't standard semver.

**Comment by ddelange** [comment]
yeah, tell me about it... [many small struggles](https://github.com/ddelange/pipgrip/pull/35/files) in pipgrip 😄

easiest way to get 1.7.1 from ~=1.0 is to parse the json output from this command:
```
pip install -q --dry-run --report - --no-deps -- maturin~=1.0
```
